packages/dds/tree/src/simple-tree/core/walkSchema.ts

```diff
@@ -3,57 +3,73 @@
  * Licensed under the MIT License.
  */
 
-import type { TreeNodeSchema } from "./treeNodeSchema.js";
+import { fail } from "@fluidframework/core-utils/internal";
+import type { AllowedTypeMetadata } from "../schemaTypes.js";
+import {
+	asTreeNodeSchemaCorePrivate,
+	type NormalizedAnnotatedAllowedTypes,
+	type TreeNodeSchema,
+} from "./treeNodeSchema.js";
 
 /**
  * Traverses all {@link TreeNodeSchema} schema reachable from `schema`, applying the visitor pattern.
  */
 export function walkNodeSchema(
 	schema: TreeNodeSchema,
+	annotations: AllowedTypeMetadata,
 	visitor: SchemaVisitor,
 	visitedSet: Set<TreeNodeSchema>,
 ): void {
 	if (visitedSet.has(schema)) {
 		return;
 	}
+
 	visitedSet.add(schema);
 
-	walkAllowedTypes(schema.childTypes, visitor, visitedSet);
+	const annotatedAllowedTypes =
+		asTreeNodeSchemaCorePrivate(schema).childAnnotatedAllowedTypes ??
+		fail("TreeNodeSchemas must implement TreeNodeSchemaCorePrivate");
```

Comment:
This fail will never happen and is unneeded.
```suggestion
	const annotatedAllowedTypes = asTreeNodeSchemaCorePrivate(schema).childAnnotatedAllowedTypes;
```